cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -2223,3 +2227,39 @@ def nims_transformations(df: DataFrame) -> DataFrame:
 def derive_overall_vaccination(df: DataFrame) -> DataFrame:
     """Derive overall vaccination status from NIMS and CIS data."""
     return df
+
+
+def add_pattern_matching_flags(df: DataFrame) -> DataFrame:
+    """Add result of various regex pattern matchings"""
+
+    # add work from home flag
+    df = assign_regex_match_result(
+        df=df,
+        columns_to_check_in=["work_main_job_title", "work_main_job_role"],
+        positive_regex_pattern=work_from_home_pattern.positive_regex_pattern,
+        negative_regex_pattern=work_from_home_pattern.negative_regex_pattern,
+        column_name_to_assign="is_working_from_home",
```

Comment:
For the function calls, we would want to use the column objects instead of assigned columns
For example: when at_school and F.col(age) < 16

This way we won't need to clean up the flag columns afterwards